rules/import-style.js

```diff
@@ -127,244 +128,382 @@ const defaultStyles = {
-			if (!(
-				node.init?.type === 'CallExpression'
-				&& node.init.callee.type === 'Identifier'
-				&& node.init.callee.name === 'require'
-			)) {
+			if (actualImportStyles.every(style => effectiveAllowedImportStyles.has(style))) {
 				return;
 			}
 
-			const assignmentTargetNode = node.id;
-			const moduleNameNode = node.init.arguments[0];
-			const moduleName = getStringIfConstant(moduleNameNode, sourceCode.getScope(moduleNameNode));
-
-			if (!moduleName) {
-				return;
-			}
+			const data = {
+				allowedStyles: new Intl.ListFormat('en-US', {type: 'disjunction'}).format([...allowedImportStyles.keys()]),
+				moduleName,
+			};
+
+			context.report({
+				node,
+				messageId: MESSAGE_ID,
+				data,
+				fix(fixer) {
+					if (!allowedImportStyles.has('namespace') || allowedImportStyles.has('named')) {
+						return;
+					}
+
+					const isImportDeclaration = node.type === 'ImportDeclaration';
+					const isVariableDeclarator = node.type === 'VariableDeclarator';
+					const isRequireCall = isCallExpression(node.init, {name: 'require'});
+					const isImportExpression = node.init?.type === 'ImportExpression';
+					const isVariableDeclaratorWithRequireCall = isVariableDeclarator && (isRequireCall || isImportExpression);
+					const isValidNode = isImportDeclaration || isVariableDeclaratorWithRequireCall;
+
+					if (!isValidNode) {
+						return;
+					}
+
+					let importedNames = [];
+					if (node.type === 'ImportDeclaration') {
+						importedNames = node.specifiers
+							.filter(s => s.type === 'ImportSpecifier' || s.type === 'ImportDefaultSpecifier')
+							.map(s => ({
+								localName: s.local.name,
+								importedName: s.type === 'ImportDefaultSpecifier' ? 'default' : s.imported.name,
+							}));
+					} else if (node.type === 'VariableDeclarator' && node.id.type === 'ObjectPattern') {
+						importedNames = node.id.properties
+							.map(p => {
+								if (p.type === 'RestElement') {
+									return {
+										localName: p.argument.name,
+										importedName: undefined,
+									};
+								}
+
+								return {
+									localName: p.value.name,
+									importedName: p.key.name,
+								};
+							});
+					}
+
+					if (importedNames.length === 0) {
+						return;
+					}
+
+					const specialCases = {
+						react: 'React',
+						'react-dom': 'ReactDOM',
+						'react-router': 'ReactRouter',
+						'react-router-dom': 'ReactRouterDOM',
+						'prop-types': 'PropTypes',
+						lodash: '_',
+						'lodash-es': '_',
+						jquery: '$',
+						'styled-components': 'styled',
+						redux: 'Redux',
+						'react-redux': 'ReactRedux',
+						axios: 'Axios',
+						moment: 'moment',
+						'date-fns': 'dateFns',
+						ramda: 'R',
+						rxjs: 'Rx',
+						vue: 'Vue',
+						angular: 'Angular',
+					};
+
+					const scope = sourceCode.getScope(node);
+					const namespaceIdentifier = specialCases[moduleName]
+						|| moduleName.replaceAll(/-./g, x => x[1].toUpperCase());
```

Comment:
This is not enough, `moduleName` can contain characters which is not allowed in identifier, eg: `.`, `#`. We need replace them first.

And it should only infer from the last part `from "./foo/bar.js"` should be `bar`

Reply:
I've added some more tests to check these use cases and updated the implementation to use `getNamespaceIdentifier` which handles creation of the namespace indentifier according to these rules.